### PR TITLE
Improvements to attribute inference attacks

### PR DESCRIPTION
**Pull request by abigailgold**
# Description

Added support for several new features:
1. Optional use of true labels in black-box attack
2. Automatic calculation of values in fit() method
3. Additional scaling method for labels/predictions
4. Additional attack model type (random forest)

## Type of change

Please check all relevant options.

- [ X] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

**Test Configuration**:
- OS: MacOS 11.6.2
- Python version: 3.8
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [ X] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my own code
- [X ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [X ] My changes generate no new warnings
- [X ] I have added tests that prove my fix is effective or that my feature works
- [X ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi @abigailgold Thank you very much for your pull request! It looks good to me and I have two questions:
- how about checking the value for `attack_model_type` for supported values in `_check_params`?
- would it be possible to move some of the new and old label processing code e.g. in `fit` to a utils function for code reuse?

**Review by beat-buesser — APPROVED** [review]
